### PR TITLE
Bugfix/123/make boundarybox smaller for all entities

### PR DESCRIPTION
**Pull request by JfriisJ**

### CODE REVIEW

File: Map/src/main/java/dk/sdu/sesem4/map/MapUtil.java [hunk 1]
```diff
@@ -98,12 +97,13 @@ public boolean[][] getNavGrid(GameData gameData) {
 	 * @param entityRectangle The entity's bounding box.
 	 * @return Whether the entity can pass through the position.
 	 */
-	protected boolean isRectangleValid(Rectangle entityRectangle, TiledMap map) {
-		float epsilon = 0.0001f;
+	protected boolean isRectangleValid(Rectangle entityRectangle) {
+		float epsilon = 2.0f;
```
Comment (The0mikkel) [inline Map/src/main/java/dk/sdu/sesem4/map/MapUtil.java:101]: This name clashes with `yAxisEpsilon`

**Review by MGertz — APPROVED** [review]
LGTM

**Review by The0mikkel — CHANGES_REQUESTED** [review]

File: Map/src/main/java/dk/sdu/sesem4/map/MapUtil.java [hunk 2]
```diff
@@ -98,12 +97,13 @@ public boolean[][] getNavGrid(GameData gameData) {
...
+		float xAxisEpsilon = 3.0f;
 		boolean bottomLeftPassible = isPositionPassable(entityRectangle.getBottomLeftCorner().plus(new Vector2(epsilon, epsilon)));
 		boolean bottomRightPassible = isPositionPassable(entityRectangle.getBottomRightCorner().plus(new Vector2(-epsilon, epsilon)));
-		boolean topLeftPassible = isPositionPassable(entityRectangle.getTopLeftCorner().plus(new Vector2(epsilon, -epsilon)));
-		boolean topRightPassible = isPositionPassable(entityRectangle.getTopRightCorner().plus(new Vector2(-epsilon, -epsilon)));
+		boolean topLeftPassible = isPositionPassable(entityRectangle.getTopLeftCorner().plus(new Vector2(epsilon * xAxisEpsilon, -epsilon * xAxisEpsilon)));
+		boolean topRightPassible = isPositionPassable(entityRectangle.getTopRightCorner().plus(new Vector2(-epsilon * xAxisEpsilon, -epsilon * xAxisEpsilon)));
 		return bottomLeftPassible && bottomRightPassible && topLeftPassible && topRightPassible;
```
Comment (The0mikkel) [inline Map/src/main/java/dk/sdu/sesem4/map/MapUtil.java:107]: This must be documented in the code. The code in itself is not self describing, and we should not rely on self describing code.

**Review by The0mikkel — CHANGES_REQUESTED** [review]

File: Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java [hunk 3]
```diff
@@ -41,13 +41,12 @@ public void testIsRectangleValid() {
 		);
 		
 		map.setCurrentMapIndex(0);
-		TiledMap tiledMap = this.map.getCurrentTiledMap();
-		
+
 		MapUtil mapUtil = new MapUtil();
 		
 		for (java.util.Map.Entry<Vector2, Boolean> entrySet : positions.entrySet()) {
```
Comment (Muhammed-als) [inline Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java:47]: Import the library instead of using it in the for loop

**Review by Muhammed-als — APPROVED** [review]
LGTM

File: Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java (hunk 3 above)
Comment (The0mikkel) [reply Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java:47]: I agree with this. It is not good practice to use full path I believe

**Review by The0mikkel — CHANGES_REQUESTED** [review]

File: Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java (hunk 3 above)
Comment (MGertz) [reply Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java:47]: This has been approved long time ago. its not part of the changes in this issue. 

Lets move on, or at least create a separate bugfix for this. as its not related to the changes here.

Comment (JfriisJ) [reply Map/src/test/java/dk/sdu/sesem4/map/MapUtilTest.java:47]: agree with MGertz, create a new issue based on that specific change.

**Pull request merged**